### PR TITLE
Add an `input_circuit(s)` method in css & qss

### PR DESCRIPTION
**Pull request by bharat-thotakura**
Allows users to output the circuit they submitted for a job in css and qss. Implementation mirrors the `compiled_circuit(s)` method.

Also ensures that compiled circuit metadata is the same as the input circuit; fixes https://github.com/Infleqtion/server-superstaq/issues/2555

Requires https://github.com/Infleqtion/server-superstaq/pull/2608.

### CODE REVIEW

File: cirq-superstaq/cirq_superstaq/job.py
Comment (epelaaez) [inline cirq-superstaq/cirq_superstaq/job.py:171]:
```suggestion
            The pre-compiled circuit.
```
or `pre-compilation`? or just `input`? I'm not sure, but pre-compile sounds weird to me.

Comment (bharat-thotakura) [reply cirq-superstaq/cirq_superstaq/job.py:171]: That is a fair point, I will use "input" as the standard between css and qss.

File: qiskit-superstaq/qiskit_superstaq/superstaq_job.py
Comment (bharat-thotakura) [reply qiskit-superstaq/qiskit_superstaq/superstaq_job.py:177]: Will update. See previous comment.

File: qiskit-superstaq/qiskit_superstaq/superstaq_job.py
Comment (epelaaez) [inline qiskit-superstaq/qiskit_superstaq/superstaq_job.py:155]:
```suggestion
            circuit_type: The kind of circuits to retrieve. Either "input_circuit" or
                "compiled_circuit".
```

Comment (epelaaez) [reply qiskit-superstaq/qiskit_superstaq/superstaq_job.py:155]: (based on how the function is called later on in this file)

Comment (bharat-thotakura) [reply qiskit-superstaq/qiskit_superstaq/superstaq_job.py:155]: Good catch! Fixed now

**Review by epelaaez — APPROVED** [review]
lgtm assuming it is not affected by https://github.com/Infleqtion/server-superstaq/pull/2636/files

**Comment by bharat-thotakura** [comment]
https://github.com/Infleqtion/server-superstaq/pull/2636/files doesn't remove returning of the input circuit so it should be fine, but I'll wait for that to be merged in first

**Pull request merged**
